properties.md

```diff
@@ -0,0 +1,140 @@
+    "value": `{"packageName":"etcd","version":"0.9.2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdCluster","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdRestore","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.gvk",
+    "value": `{"group":"etcd.database.coreos.com","kind":"EtcdBackup","version":"v1beta2"}`,
+},
+{
+    "type":  "olm.label",
+    "value": `{"label":"testlabel"}`,
+}
+```
+
+Properties intentionally closely align with the format for [`dependencies.yaml`](https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md#bundle-dependencies).
+
+Today, all properties are derived from other fields in a bundle. There is no way to write a new property `type` for a consumer (say, a newer version of the resolver, or an external tool that processes sets of bundles).
+
+### Other consumers
+
+Catalog and Bundles are consumed in many other contexts than the resolver. The package-server component presents catalog content for the purpose of selecting packages and channels, for example. 
+
+### Goals
+
+- Define the format for explicitly adding properties to a bundle.
+
+## Proposal
+
+A new file is accepted in the `metadata` folder for a bundle: `properties.(yaml|json)`
+
+Here is an example of `properties.yaml`:
+
+```yaml=
+properties:
+- type: olm.package
+  value:
+    packageName: prometheus
+    version: "0.27.0"
+- type: olm.gvk
+  value:
+    group: etcd.database.coreos.com
+    kind: EtcdCluster
+    version: v1beta2
+- type: olm.label
+  value: "LTS"
+```
+
+If properties are defined, they take precendence over the values derived from the bundle contents. In the above example, the `version` of the bundle will be considered `0.27.0` regardless of what the `ClusterServiceVersion`'s `spec.version` field indicates.
+
+### Validate properties
+
+Property validation is the domain of the consumer, not the producer. 
+
+However, `opm` will ship with validation for all properties that the OLM resolver understands, and they will be validated as part of the `opm alpha bundle validate` command and the `operator-sdk` bundle validation. 
+
+A future enhancement will discuss a formalized way to discover / publish validation rules for a given property `type`.
+
+### Generate default properties
+
+`opm` and `operator-sdk` will be capable of scaffolding the set of inferred properties from a given bundle. `opm` today infers a set of properties based on bundle content and stores them in the registry.
+
+### Initial Resolver-Consumer Properties
+
+This is a list of properties that the resolver consumer will read and understand.
+
+- `olm.package`: The package name and bundle version. Used as a target of version range dependencies.
+- `olm.gvk`: The GVK of a provided api. Used as the target of a required API dependency.
+- `olm.replaces`: The name of another bundle that this one replaces
```

Comment:
This doesn't align exactly with the declarative config spec, which currently defines an `olm.channel` property whose value contains the name of a channel and the bundle that this bundle replaces (if any).

So effectively, `replaces` is channel-specific as defined in the EP spec.

I think we need to de-conflict one way or the other.

Reply:
I was imagining channel and replaces were more or less orthogonal, so that:

```
olm.channel: alpha
olm.replaces: my-prev-operator.v1
```

```
olm.channel: alpha
olm.channel: beta
olm.replaces: my-pre-operator.v1
```

```
olm.channel: alpha
olm.replaces: my-pre-operator.v1
---
olm.channel: beta
olm.replaces: my-pre-operator.v1
```

are all valid. but I think we can make either way work and adjust in the future if desired (kind of the point of properties - long term we've even discussed `replaces` being a query over all other bundles)

Reply:
Using `alpha` and `beta` is actually helpful to illustrate the benefit of channel-specific replaces.

Say alpha looks like `A <- B <- C <- D <- E <- F`
And beta looks like `B <- D`

And I want to promote `F` to the beta channel.

`E` doesn't exist in `beta`, so I would need to say:
```yaml
olm.channel: beta
olm.replaces: D
```

But that would result in `F` having:
```
olm.channel: alpha
olm.channel: beta
olm.replaces: E
olm.replaces: D
```

Are multiple replaces allowed? If so, how do I know which `replaces` goes with which channel? Do they both go with both channels? If so, adding `olm.replaces: D` also affects the graph for `alpha` because it would look like:
```
A <- B <- C <- D <- E <- F
               ^        /
                \______/
```

Reply:
I updated to match with declarative config for now, we can revisit as needed by adding properties.